### PR TITLE
VS Code - Produtividade Infinita

### PR DESCRIPTION
**Pull request by felipe-cabrera**
## What does this PR do?
Add resource(s)

### Description 

The VS Code Produtividade Infinita (infinite productivity) ebook is a compilation of tips and teachings on how to improve your productivity as a programmer using VS Code

### Why is this valuable (or not)?

You can improve your producitivity

### How do we know it's really free?

I'm the author, it's 100% free and it's hosted on GIT Hub :)

The e-book is made with MarkDown.

### For book lists, is it a book? For course lists, is it a course? etc.

It's a Book.

## Checklist:
- [X] Read our [contributing guidelines](/CONTRIBUTING.md)
- [X] Search for duplicates.
- [X] Include author(s) and platform where appropriate.
- [X] Put lists  in alphabetical order, correct spacing.
- [X] Add needed indications (PDF, access notes, under construction)

## Followup

- Check the output of Travis-CI for linter errors!

### CODE REVIEW

**Comment by eshellman** [comment]
It's not so helpful to add a category for a single resource. Try to find another category. Or in this case, make the category broader to attract more items - maybe "Visual Studio"?

**Comment by felipe-cabrera** [comment]
Hi Eric, thanks for your feedback :)

Visual Studio very is different from Visual Studio Code, that's why i cant change it to a broader version like you said.

Currently, VS Code is one of the most used Code Editor (as Stackoverflow Developers Survey found).

VS Code works for almost all programming languages. So, on my e-book i compiled more than 250 tips about VS Code its self and more than 100 extensions for a lot of languagens, like C#, Python, Java, JavaScript, Html, C++, PHP ...

So, it's unfair if i put it only in Python because it's my favorite language :) 

But if you have any tip to change it, let me know

**Comment by eshellman** [comment]
OK, so it's more like emacs and vim, which have categories in our english list. Maybe those could be grouped under "development tools" or "programming tools"?  Also, did you mean to add to the courses list instead of the books list?

**Comment by davorpa** [comment]
I agree with @eshellman. Resource needs moving to [programming-books-pt_BR.md](https://github.com/EbookFoundation/free-programming-books/blob/master/books/free-programming-books-pt_BR.md) 😉

In this listing I propose join Vim category as:
```
## IDE / Editores
   - Your book
   ### Vim
      - vim books 1
      - ...
```
And move/grouping all in `Agnósticos` section

```
## Agnósticos
   ...
   ### IDE / Editores
      - Your book
      #### Vim
         - vim book 1....
         - ...
   ...
```

File: free-courses-pt_BR.md [hunk 1]
```diff
@@ -285,6 +287,8 @@
 
 * [Aprendendo Swift do Iniciante ao Avançado. (Mac e Windows)](https://www.udemy.com/aprendendoswift3/) - Lucas Alencar (Udemy)
 
+### VS Code
+* [VS Code Produtividade infinita](https://github.com/bylearn/VS-Code-Produtividade-Infinita)
```
Comment (davorpa) [inline free-courses-pt_BR.md:291]: Move resource proposed. See: https://github.com/EbookFoundation/free-programming-books/pull/5193#issuecomment-886314423

File: free-courses-pt_BR.md [hunk 2]
```diff
@@ -285,6 +287,8 @@
 
 * [Aprendendo Swift do Iniciante ao Avançado. (Mac e Windows)](https://www.udemy.com/aprendendoswift3/) - Lucas Alencar (Udemy)
 
+### VS Code
```
Comment (davorpa) [inline free-courses-pt_BR.md:290]: Revert change proposed. See: https://github.com/EbookFoundation/free-programming-books/pull/5193#issuecomment-886314423

File: free-courses-pt_BR.md [hunk 3]
```diff
@@ -34,9 +34,11 @@
 * [Sass](#sass)
 * [Shell](#shell)
 * [Swift](#swift)
+* [VS Code](#vscode)
 * [Vue.js](#vuejs)
 
 
+
```
Comment (davorpa) [inline free-courses-pt_BR.md:41]: Revert extra newline.

File: free-courses-pt_BR.md [hunk 4]
```diff
@@ -34,9 +34,11 @@
 * [Sass](#sass)
 * [Shell](#shell)
 * [Swift](#swift)
+* [VS Code](#vscode)
```
Comment (davorpa) [inline free-courses-pt_BR.md]: Revert change. See proposition: https://github.com/EbookFoundation/free-programming-books/pull/5193#issuecomment-886314423

**Review by davorpa — CHANGES_REQUESTED** [review]

**Review by davorpa — CHANGES_REQUESTED** [review]
Bump review. @felipe-cabrera remember apply suggestions before sync your repo :wink: 

Conflicts are produced by a repo reorganization (course files was moved to courses folder `free-courses-pt_BR.md`~>`courses/ree-courses-pt_BR.md`) made on March 2021.

https://github.com/EbookFoundation/free-programming-books/pull/5193#issuecomment-886314423

```diff
diff --git a/free-courses-pt_BR.md b/free-courses-pt_BR.md
--- free-courses-pt_BR.md
+++ free-courses-pt_BR.md
@@ -33,13 +33,11 @@
 * [Ruby](#ruby)
 * [Sass](#sass)
 * [Shell](#shell)
 * [Swift](#swift)
-* [VS Code](#vs-code)
 * [Vue.js](#vuejs)
 
 
-
 ### Android
 
 * [Android Start - Curso Android - Java Kotlin e Flutter](https://www.udemy.com/android-start-curso-android-basico-java-kotlin-e-flutter/) - Buno de Libero Ferreira (Udemy)
 * [Desenvolvedor Android Iniciante](https://www.udemy.com/desenvolvedor-android-iniciante/) - Gabriel Ferrari, Adriano Sacardo (Udemy)
@@ -286,10 +284,8 @@
 ### Swift
 
 * [Aprendendo Swift do Iniciante ao Avançado. (Mac e Windows)](https://www.udemy.com/aprendendoswift3/) - Lucas Alencar (Udemy)
 
-### VS Code
-* [VS Code Produtividade infinita](https://github.com/bylearn/VS-Code-Produtividade-Infinita)
 
 ### Vue.js
 
 * [Minicurso: Vue.js - O basicão](https://evolutio.io/curso/minicurso_vuejs) - Tony Lâmpada (Evolutio)
```
```diff
diff --git a/free-programming-books-pt_BR.md b/free-programming-books-pt_BR.md
--- free-programming-books-pt_BR.md
+++ free-programming-books-pt_BR.md
@@ -1,7 +1,8 @@
 ### Índice
 
 * [Agnósticos](#agnosticos)
+  * [IDE - Editores](#ide---editores)
   * [Programação](#programação)
   * [Sistemas Operacionais](#sistemas-operacionais)
 * [Android](#android)
 * [C](#c)
@@ -43,13 +44,27 @@
 * [R](#r)
 * [RegEx](#regex)
 * [Ruby](#ruby)
 * [Shell Script](#shell)
-* [Vim](#vim)
 
 
 ### Agnósticos
 
+#### IDE - Editores
+
+* [Visual Studio Code: Produtividade infinita](https://github.com/bylearn/VS-Code-Produtividade-Infinita) - Felipe Cabrera Ribeiro dos Santos
+
+
+##### Vim
+
+* [Editor VI - Guia de Referência](http://aurelio.net/curso/material/vim-ref.html) - Aurelio Marinho Jargas
+* [O Editor de Texto Vim](https://code.google.com/p/vimbook) - Sérgio Luiz Araújo Silva et al.
+* [Vim Avançado](http://aurelio.net/vim/vim-avancado.txt) - Aurelio Marinho Jargas
+* [Vim Básico](http://aurelio.net/vim/vim-basico.txt) - Aurelio Marinho Jargas
+* [Vim Médio](http://aurelio.net/vim/vim-medio.txt) - Aurelio Marinho Jargas
+* [Vim para Noobs](http://woliveiras.com.br/vimparanoobs/) - William Oliveira Souza
+
+
 #### Programação
 
 * [Introdução a Algoritmos e Programação](http://www.ferrari.pro.br/home/documents/FFerrari-CCechinel-Introducao-a-algoritmos.pdf) - Fabricio Ferrari e Cristian Cechinel (PDF)
 * [Lógica de Programação para iniciantes](https://dicasdeprogramacao.com.br/download/ebook-logica-de-programacao-para-iniciantes.pdf) - Gustavo Furtado de Oliveira Alves (PDF)
@@ -339,14 +354,4 @@
 
 * [Introdução ao Shell Script](http://aurelio.net/shell/apostila-introducao-shell.pdf) - Aurelio Marinho Jargas (PDF)
 * [Pequena introdução ao linux e ao Shell Script](https://www.telecom.uff.br/pet/petws/downloads/apostilas/LINUX.pdf) (PDF)
 * [Shell Script do Zero](http://metamorphoselinux.net/Shell_Script_do_Zero.pdf) (PDF)
-
-
-### Vim
-
-* [Editor VI - Guia de Referência](http://aurelio.net/curso/material/vim-ref.html) - Aurelio Marinho Jargas
-* [O Editor de Texto Vim](https://code.google.com/p/vimbook) - Sérgio Luiz Araújo Silva et al.
-* [Vim Avançado](http://aurelio.net/vim/vim-avancado.txt) - Aurelio Marinho Jargas
-* [Vim Básico](http://aurelio.net/vim/vim-basico.txt) - Aurelio Marinho Jargas
-* [Vim Médio](http://aurelio.net/vim/vim-medio.txt) - Aurelio Marinho Jargas
-* [Vim para Noobs](http://woliveiras.com.br/vimparanoobs/) - William Oliveira Souza

```

**Comment by eshellman** [comment]
please resolve conflicts.

**Review by davorpa — APPROVED** [review]
Review of stale PR with conflicts moved to #5514

**Pull request closed without merging**